src/Language/Pirouette/QuasiQuoter/Syntax.hs

```diff
@@ -276,35 +421,133 @@ parseBinder binder parseVars parseBody = do
   b <- parseBody
   return $ foldr (uncurry binder) b vars
 
+-- | Parse a declaration of a type variable and its kind. The kind is `*` by
+-- default if omitted.
+-- E.g. `(a : * -> * -> *)`
+-- E.g. `(a : *)` and `a` are equivalent
+parseTypeVarKind :: forall lang. LanguageParser lang => Parser (String, SystF.Kind)
+parseTypeVarKind = parens p <|> p
```

Comment:
Is the `parens` really necessary here? Shouldn't it be the responsibility of the caller to add `parens parseTypeVarKind` in the places it needs it?

Reply:
It was used in all call sites of `parseTypeVarKind` so I moved it in the parser definition but I conceptually agree and moved back `parens` to call sites.